app/models/targeting.rb

```diff
@@ -10,7 +10,9 @@ class Targeting < ActiveRecord::Base
 
   has_many :targeting_hosts, :dependent => :destroy
   has_many :hosts, :through => :targeting_hosts
+  # rubocop:disable Rails/HasManyOrHasOneDependent
```

Comment:
Do you have any insights on why this has happened? Is there a risk this would affect behavior in other places that we have the `:dependent` specified? Why is it raising this? Why it complains it wants to crate something, when we are actually deleting

```
ActiveRecord::RecordNotSaved: You cannot call create unless the parent is saved
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:11:in `block in transaction'
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:5:in `transaction'
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:11:in `block in transaction'
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:5:in `transaction'
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:11:in `block in transaction'
    test_after_commit (1.1.0) lib/test_after_commit/database_statements.rb:5:in `transaction'
    /var/lib/workspace/workspace/test_plugin_pull_request/database/mysql/label/fast/ruby/2.2/plugin/app/models/concerns/foreman_remote_execution/foreman_tasks_cleaner_extensions.rb:29:in `delete_job_invocations'
    /var/lib/workspace/workspace/test_plugin_pull_request/database/mysql/label/fast/ruby/2.2/plugin/app/models/concerns/foreman_remote_execution/foreman_tasks_cleaner_extensions.rb:25:in `delete_tasks'
    foreman-tasks (0.10.4) lib/foreman_tasks/cleaner.rb:104:in `delete'
    /var/lib/workspace/workspace/test_plugin_pull_request/database/mysql/label/fast/ruby/2.2/plugin/app/models/concerns/foreman_remote_execution/foreman_tasks_cleaner_extensions.rb:12:in `delete' (ActiveRecord::RecordNotSaved)
/usr/local/rvm/gems/ruby-2.2.5@test_plugin_pull_request-1/gems/activerecord-4.2.9/lib/active_record/associations/collection_association.rb:497    
```

Reply:
When I tried to set it to `:dependent => :nullify` I got an error about modifying an already destroyed record, so probably `JobInvocation` got destroyed, triggered destroy callbacks on its `Targeting` which tried to modify already destroyed `JobInvocation`. It is quite possible the same thing happens with `:dependent => :destroy`.

I'll take some time tomorrow to go over all the changes and check it is safe

Reply:
Any update on this?